src/components/cubes.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Cube from './cube';
+
+import useColumnOptions from './use/column-options';
+import CubeColumnChooser from './cube-column-chooser';
+
+import './cubes.pcss';
+
+export function Cubes({ app }) {
+  const [count, setCount] = useState(1);
+  const [cubeList, setCubeList] = useState([]);
+  const [, forceUpdate] = useState(null);
+  const addButtonRef = useRef(null);
+
+  const selectableColumns = useColumnOptions(app);
+  const addOpen = useRef(false); // Start with add dialog open
+
+  function addCube(column) {
+    if (column) {
+      const newCube = { id: count, initialColumns: [column] };
+      setCount(count + 1);
```

Comment:
I can't really see why we'd need `count` as a separate state, use `cubeList.length`?